### PR TITLE
[alerts] Reduce the threshold for WebAppServicesCrashlooping

### PR DESCRIPTION
**Pull request by geropl**
and add an internal version with even lower threshold for further improvements

Situation that we are trying to detect:
image

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/WEB-607/bridge-crash-incident-crashloop-alert-did-not-fire

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: operations/observability/mixins/meta/rules/server.yaml [hunk 1]
```diff
@@ -98,17 +98,32 @@ spec:
...
+
+    - alert: WebAppServicesCrashloopingInternal
+      # Reasoning: alert if any pod is stuck in crashlooping more than 1 minute.
+      # Used for fine tuning the alert above (WebAppServicesCrashlooping)!
+      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster, pod) > 1
+      # Let's be more ambitious than 5m
+      for: 1m
+      labels:
+        # sent to the team internal channel until we can propagate these values to CRITICAL
         severity: warning
```
Comment (geropl) [inline operations/observability/mixins/meta/rules/server.yaml:121]: Make the internal one even stricter: This applies to _some_ cases over the last months, but those feel too often, so we should fix the root cause. :+1:

File: operations/observability/mixins/meta/rules/server.yaml [hunk 2]
```diff
@@ -98,17 +98,32 @@ spec:
...
+        team: webapp
+      annotations:
+        runbook_url: https://github.com/gitpod-io/runbooks/blob/main/runbooks/WebAppServicesCrashlooping.md
+        summary: Pod is crash looping in {{ $labels.cluster }}.
+        description: Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is stuck in crashlooping for at least 5 minutes
+
+    - alert: WebAppServicesCrashloopingInternal
+      # Reasoning: alert if any pod is stuck in crashlooping more than 1 minute.
+      # Used for fine tuning the alert above (WebAppServicesCrashlooping)!
+      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster, pod) > 1
```
Comment (easyCZ) [inline operations/observability/mixins/meta/rules/server.yaml:116]:
```suggestion
      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster, pod) > 0
```
?

Comment (geropl) [reply operations/observability/mixins/meta/rules/server.yaml:116]: Whoops, that's what I wanted to do - wrong operator! :grimacing:

File: operations/observability/mixins/meta/rules/server.yaml [hunk 3]
```diff
@@ -98,17 +98,32 @@ spec:
         dashboard_url: https://grafana.gitpod.io/d/6581e46e4e5c7ba40a07646395ef7b23/kubernetes-compute-resources-pod?var-cluster={{ $labels.cluster }}&var-namespace=default
 
     - alert: WebAppServicesCrashlooping
-      # Reasoning: alert if any pod is restarting more than 3 times / 5 minutes.
-      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster) > 3
+      # Reasoning: alert if any pod is stuck in crashlooping more than 5 minute.
+      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster, pod) > 1
```
Comment (easyCZ) [inline operations/observability/mixins/meta/rules/server.yaml:102]:
```suggestion
      expr: sum(increase(kube_pod_container_status_restarts_total{container!="POD", pod=~"(content-service|dashboard|db|messagebus|proxy|server|ws-manager-bridge|usage)-.*"}[5m])) by (cluster, pod) > 0
```
Also here?

File: operations/observability/mixins/meta/rules/server.yaml [hunk 4]
```diff
@@ -98,17 +98,32 @@ spec:
...
+      for: 1m
+      labels:
+        # sent to the team internal channel until we can propagate these values to CRITICAL
         severity: warning
         team: webapp
       annotations:
         runbook_url: https://github.com/gitpod-io/runbooks/blob/main/runbooks/WebAppServicesCrashlooping.md
         summary: Pod is crash looping in {{ $labels.cluster }}.
-        description: Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is restarting {{ printf "%.2f" $value }} times / 5 minutes
+        description: "FINE TUNE: Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is stuck in crashlooping for at least 1 minutes"
```
Comment (easyCZ) [inline operations/observability/mixins/meta/rules/server.yaml:126]:
```suggestion
        description: "FINE TUNE: Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container }}) is stuck in crashlooping for at least 1 minute"
```

**Review by easyCZ — APPROVED** [review]

**Pull request merged**